src/vm.rs

```diff
@@ -0,0 +1,893 @@
+use std::{
+    fs::File,
+    io::{self, Read, Write},
+    time::Duration,
+};
+
+use timeout_readwrite::{TimeoutReadExt, TimeoutReader};
+
+const KEYBOARD_TIMEOUT: u64 = 10;
+
+use crate::{
+    instructions::{ConditionFlag, Instruction, REGISTER_COUNTER, Register, TrapCode, VMError},
+    terminal::KeyboardAddresses,
+};
+
+#[derive(Debug, Clone, Copy)]
+pub struct VirtualMachine {
+    memory: [u16; u16::MAX as usize],
+    registers: [u16; REGISTER_COUNTER],
+    running: bool,
+}
+
+impl VirtualMachine {
+    /// starts the visual machine with everything with 0s
```

Comment:
nit: typo
```suggestion
    /// starts the virtual machine with everything with 0s
```

It seems to happen in a many doc comments here

Reply:
Fixed